corgi/tasks/pulp.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from celery_singleton import Singleton
+
+from config.celery import app
+from corgi.collectors.pulp import Pulp
+from corgi.core.models import (
+    Channel,
+    ProductComponentRelation,
+    ProductStream,
+    ProductVariant,
+)
+from corgi.tasks.brew import fetch_modular_builds
+from corgi.tasks.common import RETRY_KWARGS, RETRYABLE_ERRORS, _create_relations
+
+logger = logging.getLogger(__name__)
+
+
+def setup_pulp_relations() -> None:
```

Comment:
This needs an ```@app.task()``` decorator if we want to run it automatically every day. Which I think we should, so that running loadproductdata manually stays identical to what celery-beat runs automatically.

Reply:
That's a very good point, thanks. I added setup_pulp_relations to the beat tasks as well.

Reply:
I also had to move setup_pulp_relations() call to the end of upload_variant_repos. That will mean it's called daily which is OK for now, but once we complete CORGI-257 we can probably just call it manually after updateproductdata once when we populate the database from scratch. In CORGI-257 we'll query the cdn_repo on demand and create collector models and entry in the relations table each time we process an erratum.